modules/order/src/OrderRefresh.php

```diff
@@ -0,0 +1,117 @@
+class OrderRefresh implements OrderRefreshInterface {
+
+  /**
+   * The order type storage.
+   *
+   * @var \Drupal\Core\Entity\EntityStorageInterface
+   */
+  protected $orderTypeStorage;
+
+  /**
+   * The chain price resolver.
+   *
+   * @var \Drupal\commerce_price\Resolver\ChainPriceResolverInterface
+   */
+  protected $chainPriceResolver;
+
+  /**
+   * The current user.
+   *
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+  protected $currentUser;
+
+  /**
+   * The order refresh processors.
+   *
+   * @var \Drupal\commerce_order\OrderProcessorInterface[]
+   */
+  protected $processors;
+
+  protected static $refreshingOrders = [];
+
+  /**
+   * Constructs a new OrderRefresh object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\commerce_price\Resolver\ChainPriceResolverInterface $chain_price_resolver
+   *   The chain price resolver.
+   * @param \Drupal\Core\Session\AccountProxyInterface $current_user
+   *   The current user.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, ChainPriceResolverInterface $chain_price_resolver, AccountProxyInterface $current_user) {
+    $this->orderTypeStorage = $entity_type_manager->getStorage('commerce_order_type');
+    $this->chainPriceResolver = $chain_price_resolver;
+    $this->currentUser = $current_user;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function addProcessor(OrderProcessorInterface $processor, $priority) {
+    $this->processors[$priority] = $processor;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function needsRefresh(OrderInterface $order) {
+    /** @var \Drupal\commerce_order\Entity\OrderTypeInterface $order_type */
+    $order_type = $this->orderTypeStorage->load($order->bundle());
+
+    // Check the order's changed time against the refresh frequency.
+    // We use time() since the REQUEST constant can become stale. This is
+    // especially true during CLI operations.
+    if (time() - $order->getChangedTime() < $order_type->getRefreshFrequency()) {
+      return FALSE;
+    }
+
+    $refresh_owner_only = ($order_type->getRefreshMode() == OrderType::REFRESH_OWNER);
+    if ($refresh_owner_only  && ($order->getOwnerId() != $this->currentUser->id())) {
+      return FALSE;
+    }
+
+    return TRUE;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function refresh(OrderInterface $order) {
+    // Do not attempt to refresh an order that is already refreshing.
+    if (!isset(self::$refreshingOrders[$order->id()])) {
+      self::$refreshingOrders[$order->id()] = TRUE;
+
+      $order->setAdjustments([]);
+      foreach ($order->getLineItems() as $line_item) {
+        $line_item->setAdjustments([]);
+        $line_item->setTitle($line_item->getPurchasedEntity()->getLineItemTitle());
+        $line_item->setUnitPrice($this->chainPriceResolver->resolve($line_item->getPurchasedEntity(), $line_item->getQuantity()));
```

Comment:
It's not readable for the resolve() call to be nested like this. One statement per line.